crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -76,13 +77,41 @@ impl Rule for OnlyUsedInRecursion {
             return;
         }
 
-        for (arg_index, arg) in function.params.items.iter().enumerate() {
-            let BindingPatternKind::BindingIdentifier(arg) = &arg.pattern.kind else {
+        for (arg_index, formal_parameter) in function.params.items.iter().enumerate() {
+            let BindingPatternKind::BindingIdentifier(arg) = &formal_parameter.pattern.kind else {
                 continue;
             };
 
             if is_argument_only_used_in_recursion(function_id, arg, arg_index, ctx) {
-                ctx.diagnostic(only_used_in_recursion_diagnostic(arg.span, arg.name.as_str()));
+                // if it is the last argument, we can remove it
+                if arg_index == function.params.items.len() - 1 {
+                    ctx.diagnostic_with_dangerous_fix(
+                        only_used_in_recursion_diagnostic(arg.span, arg.name.as_str()),
+                        |fixer| {
+                            let mut fix = fixer.new_fix_with_capacity(
+                                ctx.semantic()
+                                    .symbol_references(
+                                        arg.symbol_id.get().expect("`symbol_id` should be set"),
+                                    )
+                                    .count()
+                                    + 1,
+                            );
+                            fix.push(Fix::delete(arg.span()));
+
+                            for reference in ctx.semantic().symbol_references(
+                                arg.symbol_id.get().expect("`symbol_id` should be set"),
+                            ) {
+                                let node = ctx.nodes().get_node(reference.node_id());
+
+                                fix.push(Fix::delete(node.span()));
+                            }
+
+                            fix
+                        },
+                    );
+                } else {
+                    ctx.diagnostic(only_used_in_recursion_diagnostic(arg.span, arg.name.as_str()));
```

Comment:
we can't fix this case, because we would have to find all the functions call sites and remove the argument